lib/specValidator.js

```diff
@@ -1,33 +1,36 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 'use strict';
 
-var fs = require('fs'),
-path = require('path'),
-util = require('util'),
-swt = require('swagger-tools').specs.v2,
-RefParser = require('json-schema-ref-parser'),
-utils = require('./util/utils'),
-cwd = process.cwd();
-
-const constraints = ['minLength', 'maxLength', 'minimum', 'maximum', 'enum', 
-  'maxItems', 'minItems', 'uniqueItems', 'multipleOf', 'pattern'];
-const xmsExamples = 'x-ms-examples', exampleInSpec = 'example-in-spec', BodyParameterValid = 'BODY_PARAMAETER_VALID',
-InternalError = 'INTERNAL_ERROR', RefNotFoundError = 'REF_NOTFOUND_ERROR',
-JsonParsingError = 'JSON_PARSING_ERROR', XmsExampleNotFoundError = 'X-MS-EXAMPLE_NOTFOUND_ERROR',
-ResponseBodyValidationError = 'RESPONSE_BODY_VALIDATION_ERROR',
-RequiredParameterNotInExampleError = 'REQUIRED_PARAMETER_NOT_IN_EXAMPLE_ERROR',
-BodyParameterValidationError = 'BODY_PARAMETER_VALIDATION_ERROR', TypeValidationError = 'TYPE_VALIDATION_ERROR',
-ConstraintValidationError = 'CONSTRAINT_VALIDATION_ERROR', StatuscodeNotInExampleError = 'STATUS_CODE_NOT_IN_EXAMPLE_ERROR';
+var util = require('util'),
+  path = require('path'),
+  fs = require('fs'),
+  Sway = require('sway'),
+  msRest = require('ms-rest'),
+  JsonRefs = require('json-refs'),
+  HttpRequest = msRest.WebResource,
+  utils = require('./util/utils'),
+  Constants = require('./util/constants'),
+  log = require('./util/logging'),
+  ResponseWrapper = require('./responseWrapper'),
+  ErrorCodes = Constants.ErrorCodes;
 
 class SpecValidator {
 
-  constructor(specPath) {
+  constructor(specPath, specInJson) {
```

Comment:
Garrett said that he will pass the entire spec as a json within the vscode extension from Autorest as a service. Hence added it for future use.